src/async_client.cpp

```diff
@@ -304,7 +305,8 @@ itoken_ptr async_client::disconnect(long timeout, void* userContext, iaction_lis
 	tok->set_action_callback(cb);
 	add_token(tok);
 
-	disconnect_options opts(int(timeout), dynamic_cast<token*>(tok.get()));
+	// TODO may truncate timeout
+	disconnect_options opts(static_cast<int>(timeout), dynamic_cast<token*>(tok.get()));
```

Comment:
The narrowing conversion of timeout variable from long to int might cause problems. We might end up with a negative timeout if the value of long timeout is greater than INT_MAX.